### PR TITLE
Controller runtime Update

### PR DESCRIPTION
**Pull request by bshephar**
This changes updates the controller-runtime and fixes the initialisation of objects to comply with the new controller-runtime standards. Namely:

- We fix the function signatures of the webhook to include `admissions.Warnings`:
https://github.com/kubernetes-sigs/controller-runtime/commit/8770b4d3b5425b156f48c971ee140ea25f9c1bfa

- We fix the initialisation of the manager to declare the webhook following the new schema:
https://github.com/kubernetes-sigs/controller-runtime/commit/91cdd8c9539c4860c5bacad36ebf75e4271f726d

- We fix the `Watches` function to comply with the new signature definition:
https://github.com/kubernetes-sigs/controller-runtime/commit/dc62ed5dadc6ea2d3e22546368b939a2db1edfb2

- Add context to EventHandlers:
https://github.com/kubernetes-sigs/controller-runtime/commit/2464a9d7af4911c6bf6e0baaaf8dc0ab08c7452c

### CODE REVIEW

File: controllers/keystoneapi_controller.go [hunk 1]
```diff
@@ -331,17 +328,17 @@ func (r *KeystoneAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
 			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
 		).
 		Complete(r)
 }
 
-func (r *KeystoneAPIReconciler) findObjectsForSrc(src client.Object) []reconcile.Request {
+func (r *KeystoneAPIReconciler) findObjectsForSrc(ctx context.Context, src client.Object) []reconcile.Request {
 	requests := []reconcile.Request{}
 
 	l := log.FromContext(context.Background()).WithName("Controllers").WithName("KeystoneAPI")
```
Comment (stuggi) [inline controllers/keystoneapi_controller.go:344]: could change this now to ` log.FromContext(ctx)...` ?

Comment (bshephar) [reply controllers/keystoneapi_controller.go]: True, and the next one down still using `context.TODO()`. Fixed both of these in the latest revision.

**Comment by bshephar** [comment]
Functional tests are passing on:
https://github.com/openstack-k8s-operators/keystone-operator/pull/365

File: api/go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/openstack-k8s-operators/keystone-operator/api
 
-go 1.19
+go 1.21
```
Comment (gibizer) [inline api/go.mod:3]: this should be 1.20

File: api/go.mod [hunk 3]
```diff
@@ -9,27 +9,26 @@ require (
 	github.com/openstack-k8s-operators/lib-common/modules/common v0.3.1-0.20240124141114-55d029e4658b
 	github.com/openstack-k8s-operators/lib-common/modules/openstack v0.3.1-0.20240124141114-55d029e4658b
 	github.com/openstack-k8s-operators/lib-common/modules/test v0.3.1-0.20240124141114-55d029e4658b
-	k8s.io/api v0.26.13
-	k8s.io/apimachinery v0.26.13
-	sigs.k8s.io/controller-runtime v0.14.7
+	k8s.io/api v0.28.3
+	k8s.io/apimachinery v0.29.1
```
Comment (gibizer) [inline api/go.mod:13]: this should be 0.28.3

File: go.mod [hunk 4]
```diff
@@ -1,41 +1,41 @@
 module github.com/openstack-k8s-operators/keystone-operator
 
-go 1.19
+go 1.21
```
Comment (gibizer) [inline go.mod:3]: 1.20

File: go.mod [hunk 5]
```diff
@@ -1,41 +1,41 @@
...
 	github.com/openstack-k8s-operators/lib-common/modules/openstack v0.3.1-0.20240124141114-55d029e4658b
 	github.com/openstack-k8s-operators/lib-common/modules/test v0.3.1-0.20240124141114-55d029e4658b
 	github.com/openstack-k8s-operators/mariadb-operator/api v0.3.1-0.20240208072109-4447f245e487
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.26.13
-	k8s.io/apimachinery v0.26.13
-	k8s.io/client-go v0.26.13
-	sigs.k8s.io/controller-runtime v0.14.7
+	k8s.io/api v0.28.3
+	k8s.io/apimachinery v0.29.1
```
Comment (gibizer) [inline go.mod:17]: 0.28.3

**Review by gibizer — CHANGES_REQUESTED** [review]

**Pull request merged**